src/Operators/diffusion.jl

```diff
@@ -0,0 +1,26 @@
+####
+#### Diffusive fluxes
+####
+
+@inline diffusive_flux_x(i, j, k, grid, c, κ_fcc) = κ_fcc * AxF(i, j, k, grid) * δx_faa(i, j, k, grid, c)
```

Comment:
I think to be useful this would probably have to dispatch on the type of `κ_fcc` (function or array, if the point is that the diffusivity is spatially varying). But do we think these functions belong in `TurbulenceClosures`? It could be best to put "basic" operators in `Operators`, and physics-specific operators that build on the fundamental ones in the places where they are used-needed.

Reply:
True, the diffusion and viscous dissipation operators probably belong in `Oceananigans.TurbulenceClosures`.

Since this PR doesn't actually touch the existing code, I think I'll keep these operators in this PR as a prototype/design and then remove all unused operators in the next PR.